### ISSUE TITLE
BUG: norm on LiveImage breaks

### ISSUE THREAD

**Issue by CJ-Wright** [NSLS-II/bluesky#725]
If one puts a norm into LiveImage things break with a division by zero error

## Expected Behavior
Norms should be ok.

## Current Behavior
```python
db.process(db[-1], LiveImage('pe1_image',
                             norm=LogNorm(vmin=1, vmax=1000),
                             cmap='viridis'))
```
yields
```python
Traceback (most recent call last):
  File "/home/christopher/dev/SHED/examples/xpd_alpha_workflow.py", line 304, in <module>
    source.emit(e)
  File "/home/christopher/dev/streams/streams/core.py", line 101, in emit
    r = parent.update(x, who=self)
  File "/home/christopher/dev/streams/streams/core.py", line 402, in update
    result = self.func(x)
  File "/home/christopher/dev/SHED/shed/event_streams.py", line 32, in wraps
    return f(*args)
  File "/home/christopher/dev/bluesky/bluesky/callbacks/core.py", line 21, in __call__
    return getattr(self, name)(doc)
  File "/home/christopher/dev/bluesky/bluesky/callbacks/broker.py", line 74, in event
    self.interpolation)
  File "/home/christopher/mc/envs/dp_dev/lib/python3.5/site-packages/xray_vision/backend/mpl/cross_section_2d.py", line 368, in __init__
    self._cb = fig.colorbar(self._im, cax=self._ax_cb)
  File "/home/christopher/mc/envs/dp_dev/lib/python3.5/site-packages/matplotlib/figure.py", line 1603, in colorbar
    cb = cbar.colorbar_factory(cax, mappable, **kw)
  File "/home/christopher/mc/envs/dp_dev/lib/python3.5/site-packages/matplotlib/colorbar.py", line 1347, in colorbar_factory
    cb = Colorbar(cax, mappable, **kwargs)
  File "/home/christopher/mc/envs/dp_dev/lib/python3.5/site-packages/matplotlib/colorbar.py", line 926, in __init__
    ColorbarBase.__init__(self, ax, **kw)
  File "/home/christopher/mc/envs/dp_dev/lib/python3.5/site-packages/matplotlib/colorbar.py", line 326, in __init__
    self.draw_all()
  File "/home/christopher/mc/envs/dp_dev/lib/python3.5/site-packages/matplotlib/colorbar.py", line 347, in draw_all
    self._process_values()
  File "/home/christopher/mc/envs/dp_dev/lib/python3.5/site-packages/matplotlib/colorbar.py", line 688, in _process_values
    b = self.norm.inverse(self._uniform_y(self.cmap.N + 1))
  File "/home/christopher/mc/envs/dp_dev/lib/python3.5/site-packages/matplotlib/colors.py", line 946, in inverse
    return vmin * ma.power((vmax / vmin), val)
ZeroDivisionError: division by zero
```

## Possible Solution
I'm not certain, maybe fix something in CrossSection?

## Your Environment
Cutting edge almost everything and then some.

**Comment by danielballan** [comment]
Do you think an error is introduced by LiveImage? It looks like a problem with LogNorm or, more likely, this usage of it.

**Comment by CJ-Wright** [comment]
I think that the error is that live image tries to create a color bar but there is no data so the norm fails. Would it be possible to create the figure or the color bar when the first event comes down?

**Comment by danielballan** [comment]
Sounds plausible. Can you reproduce that with an SSCCE with matplotlib alone?

**Comment by CJ-Wright** [comment]
It seems that this is a mpl problem. It seems that `imshow` on an empty list with a log norm causes problems.

**Comment by tacaswell** [comment]
fixed upstream, will be 2.1
